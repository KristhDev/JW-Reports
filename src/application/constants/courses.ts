export const coursesMessages = {
    EXPORTED_SUCCESS: 'Has exportado tus cursos correctamente. El archivo se encuentra en su carpeta de descargas.',
    PERSON_NAME_REQUIRED: 'El nombre de la persona es requerido.',
    PERSON_NAME_MIN_LENGTH: 'El nombre de la persona debe tener al menos 2 caracteres.',
    PERSON_ABOUT_REQUIRED: 'La información de la persona es requerida.',
    PERSON_ABOUT_MIN_LENGTH: 'La información de la persona debe tener al menos 10 caracteres.',
    PERSON_ADDRESS_REQUIRED: 'La dirección es requerida.',
    PERSON_ADDRESS_MIN_LENGTH: 'La dirección debe tener al menos 10 caracteres.',
    PUBLICATION_REQUIRED: 'La publicación de estudio no puede estar vacía.',
    PUBLICATION_MIN_LENGTH: 'La publicación de estudio debe tener al menos 5 caracteres.',
    UNSELECTED: 'No hay un curso seleccionado.',
    UNSELECTED_FINISH_OR_START: 'No pudes terminar o comenzar de nuevo un curso suspendido.',
    UNSELECTED_DELETE: 'No hay un curso seleccionado para eliminar.',
    UNSELECTED_UPDATE: 'No hay un curso seleccionado para actualizar.',
    FINISHED: 'No puedes suspender o renovar un curso terminado.',
    SUSPENDED_SUCCESS: 'Has suspendido el curso correctamente.',
    RENEW_SUCCESS: 'Has renovado el curso correctamente.',
    DELETED_SUCCESS: 'Has eliminado el curso correctamente.',
    FINISHED_SUCCESS: 'Has terminado el curso correctamente.',
    RESTARTED_SUCCESS: 'Has comenzado de nuevo el curso correctamente.',
    ADDED_SUCCESS: 'Has agregado el curso correctamente.',
    UPDATED_SUCCESS: 'Has actualizado el curso correctamente.',
}