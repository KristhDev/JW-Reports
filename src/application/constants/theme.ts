import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];