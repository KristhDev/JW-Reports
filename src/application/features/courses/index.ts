export {
    default as coursesReducer,
    INIT_COURSE,
    COURSES_INITIAL_STATE,
    addCourse,
    addCourses,
    addLastLessonInCourse,
    clearCourses,
    removeCourse,
    removeCourses,
    replaceLastLessonInCourse,
    setCourseFilter,
    setCourses,
    setCoursesPagination,
    setCoursesScreenHistory,
    setHasMoreCourses,
    setIsCourseDeleting,
    setIsCourseLoading,
    setIsCoursesExporting,
    setIsCoursesLoading,
    setRefreshCourses,
    setSelectedCourse,
    updateCourse,
    updateLastLessonInCourse
} from './slice';

export * from './types';