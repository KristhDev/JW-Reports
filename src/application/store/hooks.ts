import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();