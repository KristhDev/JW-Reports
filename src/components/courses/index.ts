export { CourseCard } from './CourseCard';
export { CoursesList } from './CoursesList';