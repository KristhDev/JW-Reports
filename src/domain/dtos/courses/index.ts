export * from './active-suspend.dto';
export * from './create.dto';
export * from './finish-start.dto';
export * from './update.dto';