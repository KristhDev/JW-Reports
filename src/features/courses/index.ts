export {
    default as coursesReducer,
    INIT_COURSE,
    INIT_LESSON,
    addCourse,
    addCourses,
    addLesson,
    addLessons,
    clearCourses,
    removeCourse,
    removeCourses,
    removeLesson,
    removeLessons,
    setCourseFilter,
    setCourses,
    setCoursesPagination,
    setCoursesScreenHistory,
    setHasMoreCourses,
    setHasMoreLessons,
    setIsCourseDeleting,
    setIsCourseLoading,
    setIsCoursesLoading,
    setIsLessonDeleting,
    setIsLessonLoading,
    setIsLessonsLoading,
    setLessons,
    setLessonsPagination,
    setRefreshCourses,
    setRefreshLessons,
    setSelectedCourse,
    setSelectedLesson,
    updateCourse,
    updateLesson
} from './slice';