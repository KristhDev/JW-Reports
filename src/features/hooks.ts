import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;