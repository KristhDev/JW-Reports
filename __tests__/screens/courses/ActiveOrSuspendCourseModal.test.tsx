import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react-native';
import hexToRgba from 'hex-to-rgba';

/* Screens */
import { ActiveOrSuspendCourseModal } from '../../../src/screens/courses';

/* Hooks */
import { useCourses, useTheme } from '../../../src/hooks';

/* Theme */
import { darkColors } from '../../../src/theme';

/* Setup */
import { onCloseMock } from '../../../jest.setup';

/* Mocks */
import { activeOrSuspendCourseMock, courseSelectedStateMock } from '../../mocks';

/* Mock hooks */
jest.mock('../../../src/hooks/useCourses.ts');
jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <ActiveOrSuspendCourseModal /> screen', () => {
    (useCourses as jest.Mock).mockReturnValue({
        state: courseSelectedStateMock,
        activeOrSuspendCourse: activeOrSuspendCourseMock
    });

    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors },
        BUTTON_TRANSLUCENT_COLOR: hexToRgba(darkColors.button, 0.50)
    });

    beforeEach(() => {
        render(
            <ActiveOrSuspendCourseModal
                isOpen
                onClose={ onCloseMock }
            />
        );
    });

    it('should to match snapshot', () => {
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('should render respective texts when selectedCourse isnt suspended', () => {

        /* Get motal text and touchable */
        const msg = screen.getByTestId('modal-text');
        const touchable = screen.getAllByTestId('button-touchable')[1];

        /**
         * Check if msg and touchable exists and contain respective values
         */
        expect(msg).toBeTruthy();
        expect(msg.props.children).toBe('¿Está seguro de suspender este curso?');
        expect(touchable).toBeTruthy();
        expect(touchable.props.children[0].props.children[1].props.children[0].props.children).toBe('SUSPENDER');
    });

    it('should call activeOrSuspendCourse when confirm button is pressed', () => {

        /* Get confirm touchable */
        const touchable = screen.getAllByTestId('button-touchable')[1];
        fireEvent.press(touchable);

        /**
         * Check if activeOrSuspendCourse is called one time with
         * respective args
         */
        expect(activeOrSuspendCourseMock).toHaveBeenCalledTimes(1);
        expect(activeOrSuspendCourseMock).toHaveBeenCalledWith(onCloseMock);
    });

    it('should call onClose when cancel button is pressed', () => {

        /* Get cancel touchable */
        const touchable = screen.getAllByTestId('button-touchable')[0];
        fireEvent.press(touchable);

        /* Check if onClose is called one times */
        expect(onCloseMock).toHaveBeenCalledTimes(1);
    });

    it('should render loader when isCourseLoading is true', () => {

        /* Mock data of useCourses */
        (useCourses as jest.Mock).mockReturnValue({
            state: {
                ...courseSelectedStateMock,
                isCourseLoading: true
            },
            activeOrSuspendCourse: activeOrSuspendCourseMock
        });

        render(
            <ActiveOrSuspendCourseModal
                isOpen
                onClose={ onCloseMock }
            />
        );

        /* Get loader and check if exists */
        const loader = screen.getByTestId('modal-loading');
        expect(loader).toBeTruthy();
    });
});