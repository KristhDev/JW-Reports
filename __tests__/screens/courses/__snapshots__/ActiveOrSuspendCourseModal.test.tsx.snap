// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <ActiveOrSuspendCourseModal /> screen should to match snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  statusBarTranslucent={true}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "justifyContent": "center",
        "zIndex": 999,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#292929",
          "borderRadius": 10,
          "justifyContent": "space-between",
          "padding": 24,
        }
      }
    >
      <Text
        style={
          {
            "color": "#B4B4B4",
            "fontSize": 19,
            "marginBottom": 0,
          }
        }
        testID="modal-text"
      >
        ¿Está seguro de suspender este curso?
      </Text>
      <View
        style={
          {
            "alignSelf": "flex-end",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "marginTop": 40,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              false,
              [
                {
                  "backgroundColor": "#C0A7E1",
                  "borderRadius": 10,
                },
                {
                  "backgroundColor": "transparent",
                  "marginRight": 5,
                },
              ],
            ]
          }
          testID="button-touchable"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "minWidth": "100%",
                  "paddingHorizontal": 20,
                  "paddingVertical": 9,
                },
                {
                  "minWidth": 0,
                  "paddingHorizontal": 12,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "color": "#121212",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "marginBottom": 2,
                      "textAlign": "center",
                    },
                    {
                      "color": "#C0A7E1",
                      "fontSize": 16,
                    },
                  ],
                ]
              }
              testID="button-text"
            >
              CANCELAR
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              false,
              [
                {
                  "backgroundColor": "#C0A7E1",
                  "borderRadius": 10,
                },
                {
                  "backgroundColor": "transparent",
                },
              ],
            ]
          }
          testID="button-touchable"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "minWidth": "100%",
                  "paddingHorizontal": 20,
                  "paddingVertical": 9,
                },
                {
                  "minWidth": 0,
                  "paddingHorizontal": 12,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "color": "#121212",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "marginBottom": 2,
                      "textAlign": "center",
                    },
                    {
                      "color": "#C0A7E1",
                      "fontSize": 16,
                    },
                  ],
                ]
              }
              testID="button-text"
            >
              SUSPENDER
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
